src/charm.py

```diff
@@ -319,6 +324,34 @@ def _get_session_ssl_cipher(self, event: ActionEvent) -> None:
 
         event.set_results({"cipher": cipher})
 
+    def _get_server_certificate(self, event: ActionEvent) -> None:
+        """Get the server certificate."""
+        certificate = "error"
+        if not self._database_config:
+            return event.set_results({"certificate": certificate})
+        try:
+            process = subprocess.run(
+                [
+                    "openssl",
+                    "s_client",
+                    "-starttls",
+                    "mysql",
+                    "-connect",
+                    f"{self._database_config['host']}:{self._database_config['port']}",
+                ],
+                capture_output=True,
+            )
+            # butchered stdout due non utf chars after the certificate
+            raw_output = process.stdout[:2800].decode("utf8")
+            matches = re.search(
+                r"^(-----BEGIN C.*END CERTIFICATE-----[,\s])", raw_output, re.MULTILINE | re.DOTALL
+            )
+            certificate = matches.group(0)
+        except Exception:
+            pass
```

Comment:
same, it might be nicer to `event.fail` (in which case the test might pick it up)